search/handlers.go

```diff
@@ -83,10 +105,89 @@ func (s *Server) handleSearchRequestProfiles(e echo.Context) error {
 		})
 	}
 
-	out, err := s.SearchProfiles(ctx, q)
+	offset, limit, err := parseCursorLimit(e)
+	if err != nil {
+		return err
+	}
+
+	typeahead := false
+	if q := strings.TrimSpace(e.QueryParam("typeahead")); q == "true" || q == "1" || q == "y" {
+		typeahead = true
+	}
+
+	out, err := s.SearchProfiles(ctx, q, typeahead, offset, limit)
 	if err != nil {
 		return err
 	}
 
 	return e.JSON(200, out)
 }
+
+func (s *Server) SearchPosts(ctx context.Context, q string, offset, size int) (*SearchPostsSkeletonResp, error) {
+	resp, err := DoSearchPosts(ctx, s.dir, s.escli, s.postIndex, q, offset, size)
+	if err != nil {
+		return nil, err
+	}
+
+	posts := []syntax.ATURI{}
+	for _, r := range resp.Hits.Hits {
+		var doc PostDoc
+		if err := json.Unmarshal(r.Source, &doc); err != nil {
+			return nil, fmt.Errorf("decoding post doc from search response: %w", err)
+		}
+
+		did, err := syntax.ParseDID(doc.DID)
+		if err != nil {
+			return nil, fmt.Errorf("invalid DID in indexed document: %w", err)
+		}
+
+		posts = append(posts, syntax.ATURI(fmt.Sprintf("at://%s/app.bsky.feed.post/%s", did, doc.RecordRkey)))
+	}
+
+	out := SearchPostsSkeletonResp{Posts: posts}
+	if len(posts) == size && (offset+size) < 10000 {
+		out.Cursor = fmt.Sprintf("%d", offset+size)
+	}
+	fmt.Println(resp.Hits.Total)
+	if resp.Hits.Total.Relation == "eq" {
+		out.HitsTotal = &resp.Hits.Total.Value
+	}
+	return &out, nil
+}
+
+func (s *Server) SearchProfiles(ctx context.Context, q string, typeahead bool, offset, size int) (*SearchActorsSkeletonResp, error) {
+	var resp *EsSearchResponse
+	var err error
+	if typeahead {
+		resp, err = DoSearchProfilesTypeahead(ctx, s.escli, s.profileIndex, q, size)
+	} else {
+		resp, err = DoSearchProfiles(ctx, s.dir, s.escli, s.profileIndex, q, offset, size)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	actors := []syntax.DID{}
+	for _, r := range resp.Hits.Hits {
+		var doc ProfileDoc
+		if err := json.Unmarshal(r.Source, &doc); err != nil {
+			return nil, fmt.Errorf("decoding profile doc from search response: %w", err)
```

Comment:
Similarly, do we want to error here or log it and keep going?

Reply:
same as above: this is a bad and very unexpected situation